### PR TITLE
JIP-21 Post creation flags posts as isMature using Azure

### PR DESCRIPTION
**Pull request by srhsrhsrhsrh**
[REMINDER: Resolve conflicts with jippy p's validation branch]

Create post uses azure's [text moderation api](https://docs.microsoft.com/en-us/azure/cognitive-services/content-moderator/client-libraries?tabs=visual-studio&pivots=programming-language-rest-api) to tag it as `isMature` or `!isMature`
- Mock data generation takes a long time since my azure free trial has a 1 TPS limit so every request (with a 1024 character limit btw) needs to have a 1 second timeout :^(
- I can add my setup to datagen later as an option (which is super jank, because of the timeout and character limit), but creating your own posts will work temporarily

TODOS:
- Before saturday, I'll add a random button somewhere (on the navbar? it can be moved later) which can be used for the demo to show toggling on/off `isMature`
- I also need to add a character limit to the post creation dialog

UPDATE: friday 6pm
- done those TODOs but I'm not going to get a re-review since they're relatively non-controversial and the isMature field name matches up a bit better with frontend (showMatureContent -> isMature)

### CODE REVIEW

File: packages/client/src/redux/slices/post-slice.ts [hunk 1]
```diff
@@ -61,6 +63,7 @@ const initialState: PostState = {
   posts: [],
   sortType: PostSortType.POPULAR,
   locationFilter: locationFilter,
+  matureFilter: false,
```
Comment (dryu99) [inline packages/client/src/redux/slices/post-slice.ts:66]: ヽ༼ຈل͜ຈ༽ﾉヽ༼ຈل͜ຈ༽ﾉヽ༼ຈل͜ຈ༽ﾉヽ༼ຈل͜ຈ༽ﾉ

**Review by dryu99 — APPROVED** [review]
ㄹㄱㅌㅁ

File: packages/client/src/redux/slices/post-slice.ts (hunk 1 above)
Comment (srhsrhsrhsrh) [reply packages/client/src/redux/slices/post-slice.ts:66]: i don't know these korean characters

**Comment by srhsrhsrhsrh** [comment]
JIP TOLD ME TO MERGE FIRST

**Pull request merged**